src/query/json.rs

```diff
@@ -77,20 +77,33 @@ impl FromQueryResult for JsonValue {
             crate::QueryResultRow::SqlxPostgres(row) => {
                 use serde_json::json;
                 use sqlx::{postgres::types::Oid, Column, Postgres, Row, Type};
+
                 for column in row.columns() {
                     let col = if !column.name().starts_with(pre) {
                         continue;
                     } else {
                         column.name().replacen(pre, "", 1)
                     };
                     let col_type = column.type_info();
+
                     macro_rules! match_postgres_type {
```

Comment:
The macro is used for matching primitives against column types, but doesn't consider the fact that postgres columns can be of type array (of a primitive).